src/node.cc

```diff
@@ -3638,6 +3641,364 @@ static void PrintHelp() {
+    "--retain_maps_for_n_gc",
+    "--trace_gc",
+    "--trace_gc_nvp",
+    "--trace_gc_ignore_scavenger",
+    "--trace_idle_notification",
+    "--trace_idle_notification_verbose",
+    "--print_max_heap_committed",
+    "--trace_gc_verbose",
+    "--trace_allocation_stack_interval",
+    "--trace_fragmentation",
+    "--trace_fragmentation_verbose",
+    "--trace_evacuation",
+    "--trace_mutator_utilization",
+    "--flush_code",
+    "--trace_code_flushing",
+    "--age_code",
+    "--incremental_marking",
+    "--incremental_marking_wrappers",
+    "--min_progress_during_incremental_marking_finalization",
+    "--max_incremental_marking_finalization_rounds",
+    "--black_allocation",
+    "--concurrent_sweeping",
+    "--parallel_compaction",
+    "--parallel_pointer_update",
+    "--trace_incremental_marking",
+    "--track_gc_object_stats",
+    "--trace_gc_object_stats",
+    "--track_detached_contexts",
+    "--trace_detached_contexts",
+    "--move_object_start",
+    "--memory_reducer",
+    "--scavenge_reclaim_unmodified_objects",
+    "--heap_growing_percent",
+    "--histogram_interval",
+    "--trace_object_groups",
+    "--heap_profiler_trace_objects",
+    "--sampling_heap_profiler_suppress_randomness",
+    "--use_idle_notification",
+    "--use_ic",
+    "--trace_ic",
+    "--tf_load_ic_stub",
+    "--tf_store_ic_stub",
+    "--native_code_counters",
+    "--always_compact",
+    "--never_compact",
+    "--compact_code_space",
+    "--cleanup_code_caches_at_gc",
+    "--use_marking_progress_bar",
+    "--zap_code_space",
+    "--trace_weak_arrays",
+    "--trace_prototype_users",
+    "--use_verbose_printer",
+    "--trace_for_in_enumerate",
+    "--trace_parse",
+    "--lazy_inner_functions",
+    "--log_regs_modified",
+    "--log_colour",
+    "--stack_trace_on_illegal",
+    "--abort_on_uncaught_exception",
+    "--trace_rail",
+    "--runtime_call_stats",
+    "--regexp_optimization",
+    "--profile_hydrogen_code_stub_compilation",
+    "--dump_counters",
+    "--map_counters",
+    "--log",
+    "--log_all",
+    "--log_api",
+    "--log_code",
+    "--log_gc",
+    "--log_handles",
+    "--log_suspect",
+    "--prof",
+    "--prof_cpp",
+    "--prof_browser_mode",
+    "--log_regexp",
+    "--logfile",
+    "--logfile_per_isolate",
+    "--perf_basic_prof",
+    "--perf_basic_prof_only_functions",
+    "--perf_prof",
+    "--perf_prof_debug_info",
+    "--perf_prof_unwinding_info",
+    "--log_internal_timer_events",
+    "--log_timer_events",
+    "--log_instruction_stats",
+    "--log_instruction_file",
+    "--log_instruction_period",
+    "--redirect_code_traces",
+    "--redirect_code_traces_to",
+    "--hydrogen_track_positions",
+    "--trace_elements_transitions",
+    "--trace_creation_allocation_sites",
+  };
+
+  for (unsigned i = 0; i < arraysize(whitelist); i++) {
+    const char* allowed = whitelist[i];
+    if (strlen(allowed) == arglen && strncmp(allowed, arg, arglen) == 0)
+      return;
+  }
```

Comment:
A couple of options I can think of include:

1) using a hash table, problem might be that cost of building the hashtable outweighs the benefits
2) have 26 whitelist arrays (or 2 dimention array) so that we can quickly cut down the list by the first letter in the option and then only search the subset for that list.  This would reduce the search time, while being 0 cost in terms of startup time.  The main drawback is that is does require us to make sure things go into the right array.